gemini/rag-engine/rag_engine_evaluation.ipynb

```diff
@@ -0,0 +1,983 @@
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5303c05f7aa6"
+      },
+      "source": [
+        "### Create a `RagCorpus` with the specified configuration."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "6fc324893334"
+      },
+      "outputs": [],
+      "source": [
+        "# See the list of current supported embedding models here: https://cloud.google.com/vertex-ai/generative-ai/docs/rag-overview#supported-embedding-models\n",
+        "# You may adjust the embedding model here if you would like.\n",
+        "embedding_mode_config = rag.EmbeddingModelConfig(\n",
+        "    publisher_model=\"publishers/google/models/text-embedding-004\"  # @param {type:\"string\", isTemplate: true},\n",
+        ")\n",
+        "\n",
+        "rag_corpus = rag.create_corpus(\n",
+        "    display_name=\"test-corpus\",\n",
+        "    description=\"A test corpus where we import the BEIR-FiQA-2018 dataset\",\n",
+        "    embedding_model_config=embedding_mode_config,\n",
+        ")\n",
+        "\n",
+        "print(rag_corpus)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "e43229f3ad4f"
+      },
+      "source": [
+        "### Load BEIR Fiqa dataset (test split)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 14,
+      "metadata": {
+        "id": "cf93d5f0ce00"
+      },
+      "outputs": [],
+      "source": [
+        "# Download and load a BEIR dataset\n",
+        "dataset = \"fiqa\"  # @param [\"arguana\", \"climate-fever\", \"cqadupstack\", \"dbpedia-entity\", \"fever\", \"fiqa\", \"germanquad\", \"hotpotqa\", \"mmarco\", \"mrtydi\", \"msmarco-v2\", \"msmarco\", \"nfcorpus\", \"nq-train\", \"nq\", \"quora\", \"scidocs\", \"scifact\", \"trec-covid-beir\", \"trec-covid-v2\", \"trec-covid\", \"vihealthqa\", \"webis-touche2020\"]\n",
+        "url = (\n",
+        "    f\"https://public.ukp.informatik.tu-darmstadt.de/thakur/BEIR/datasets/{dataset}.zip\"\n",
+        ")\n",
+        "out_dir = \"datasets\"\n",
+        "data_path = util.download_and_unzip(url, out_dir)\n",
+        "\n",
+        "# Load the dataset\n",
+        "corpus, queries, qrels = GenericDataLoader(data_folder=data_path).load(split=\"test\")\n",
+        "print(\n",
+        "    f\"Successfully loaded the {dataset} dataset with {len(corpus)} files and {len(queries)} queries!\"\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "j0sHOGwdTDXZ"
+      },
+      "source": [
+        "### Define helper function for processing dataset."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 36,
+      "metadata": {
+        "id": "_k4EWn3TTWVh"
+      },
+      "outputs": [],
+      "source": [
+        "def convert_beir_to_rag_corpus(corpus, output_dir):\n",
```

Comment:
These can be good utility functions to have, but it could be much easier for the user to have this already loaded in a public GCS bucket. We have a bucket for this repo `gs://github-repo/`, let me know if you want me to upload these files to it.

Reply:
Hi Holt, yes agreed that it would be much easier for the users. Could you help us upload the corpus directory to the public GCS bucket? (I just shared the drive link to you).

I will update the Colab correspondingly to let user know that if they don't want to import themselves (to save time), they could use the public bucket (`gs://github-repo/`)

Reply:
Updated the Colab with an option to use public GCS bucket for data ingestion, as discussed.